tests/common/fixtures/elasticsearch.py

```diff
@@ -29,34 +34,18 @@ def es_connect():
 def init_elasticsearch(request):
     """Initialize the test (old) Elasticsearch index once per test session."""
     client = _es_client()
+    # Initialize the test search index.
+    search.init(es_client)
     """Connect to the newer v6.x instance of Elasticsearch once per test session"""
     es_connect()
 
-    def maybe_delete_index():
-        """Delete the test index if it exists."""
-        if client.conn.indices.exists(index=ELASTICSEARCH_INDEX):
-            client.conn.indices.delete(index=ELASTICSEARCH_INDEX)
-
-    # Delete the test search index at the end of the test run.
-    request.addfinalizer(maybe_delete_index)
-
-    # Delete the test search index at the start of the run, just in case it
-    # was somehow left behind by a previous test run.
-    maybe_delete_index()
-
-    # Initialize the test search index.
-    search.init(client)
 
+    yield es_client
```

Comment:
This code doesn't look to me like it is equivalent with what was there before. We are no longer calling removing the index at the start of the run, in case it already exists for some reason.

Reply:
I question why/if we actually need to do that. Since we delete the index after every use of the connection is there going to be a case where the index hasn’t been deleted?

Reply:
If the test runner crashes or gets killed during execution, the index won't be deleted. This may be rare but my experience has been that if a test depends on persistent resources, it always needs to ensure they are in a clean state at the start of a run and can't rely only on cleaning them up at the end because there is a long tail of things that can prevent cleanup code from running.

Reply:
Ah ok. That makes sense.